src/instrumentsscene/view/instrumentlistmodel.cpp

```diff
@@ -244,12 +244,18 @@ void InstrumentListModel::loadGroups()
     m_groups = acceptedGroups;
     emit groupsChanged();
 
-    if (m_saveCurrentGroup) {
-        emit currentGroupIndexChanged();
-        return;
+    bool currentGroupInNewGenre = false;
+    for (const InstrumentGroup* group : m_groups) {
+        if (group->id == m_currentGroupId) {
+            currentGroupInNewGenre = true;
+            break;
+        }
+    }
+    if (!currentGroupInNewGenre && !isSearching()) {
+        setCurrentGroupIndex(0);
     }
 
-    doSetCurrentGroup(NONE_GROUP_ID);
+    emit currentGroupIndexChanged();
```

Comment:
Is this `emit` necessary? the signal is already emitted in `doSetCurrentGroup`, isn't it?

Reply:
The thing is, when you change the genre, `m_currentGroupId` remains same. But there is a chance that the index can change which is why emit that signal. In the following case, the index of "Keyboards" change:
[Screencast from 22-07-23 02:19:19 PM IST.webm](https://github.com/musescore/MuseScore/assets/88136054/fe1a1f76-0c9c-4dfb-9b69-13e4e511c905)

So you need to emit the changed signal, so the new index can be calculated